internal/environment/checks.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package environment
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/corazawaf/coraza/v3/debuglog"
+)
+
+// CheckFSAccess is a helper function to check if the WAF has access to the filesystem
+func CheckFSAccess(logger debuglog.Logger, tmpDir string) error {
+	if HasAccessToFS {
+		file, err := os.CreateTemp(tmpDir, "checkfsfile")
+		if err != nil {
+			return fmt.Errorf("create file: %w", err)
+		}
+		defer file.Close()
+
+		_, err = file.WriteString("check fs data")
+		if err != nil {
+			return fmt.Errorf("write to file: %w", err)
+		}
+
+		err = os.Remove(file.Name())
```

Comment:
not sure about this. Here we are removing the file but lines before we defer on closing. Also, I think if we truly remove, a file and forget is enough.

Reply:
Thanks, moved into the defer function as a fire and forget call